### PR TITLE
IBX-182: Document templating layout parts

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-182
| Versions      | 3.x

### CODE REVIEW

**Review by damianz5 — APPROVED** [review]

File: docs/guide/content_rendering/layout/add_menu.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+To create a menu that contains a specific set of Content items, for example all content under the root Location, use a [Query Type](../queries_and_controllers/content_queries.md).
+
+First, create a custom `MenuQueryType` that queries for all items you want to have in the menu:
+
+``` php hl_lines="15 16 28"
+[[= include_file('code_samples/front/layouts/menu/src/QueryType/MenuQueryType.php') =]]
+```
+
+In this case, it queries for all visible children of Location `2`, the root Location (lines 15-16)
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:23]:
```suggestion
In this case, it queries for all visible children of Location `2`, the root Location, (lines 15-16)
```

File: docs/guide/content_rendering/layout/add_menu.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+To select items that should be rendered in the menu, you can use one of the following ways:
+
+- create a [query](#render-menu-using-a-query)
+- create a [MenuBuilder](#create-a-menubuilder)
+
+## Render menu using a query
+
+To create a menu that contains a specific set of Content items, for example all content under the root Location, use a [Query Type](../queries_and_controllers/content_queries.md).
+
+First, create a custom `MenuQueryType` that queries for all items you want to have in the menu:
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:17]:
```suggestion
First, create a custom `MenuQueryType` that queries for all items that you want in the menu:
```

File: docs/guide/content_rendering/layout/add_menu.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+``` php hl_lines="15 16 28"
+[[= include_file('code_samples/front/layouts/menu/src/QueryType/MenuQueryType.php') =]]
+```
+
+In this case, it queries for all visible children of Location `2`, the root Location (lines 15-16)
+and renders them in order according to their Location priority.
+
+The Query Type has the name `Menu` (line 28), so you can use it in the template to render the menu.
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:26]:
```suggestion
The Query Type has the name `Menu` (line 28).
You can use it in the template to render the menu.
```
or
```suggestion
The Query Type has the name `Menu` (line 28), and you can use it in the template to render the menu.
```
(There is no cause effect relationship between the two statements)

File: docs/guide/content_rendering/layout/add_menu.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+which renders the individual items of the menu:
+
+``` html+twig
+[[= include_file('code_samples/front/layouts/menu/templates/themes/my_theme/pagelayout_menu.html.twig') =]]
+```
+
+## Create a MenuBuilder
+
+To make a more configurable menu, where you select the specific items to render,
+use the [KNPMenuBundle](https://github.com/KnpLabs/KnpMenuBundle) which is installed together with the product.
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:43]:
```suggestion
use the [KNPMenuBundle](https://github.com/KnpLabs/KnpMenuBundle) that is installed together with the product.
```

File: docs/guide/content_rendering/layout/add_menu.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+To use it, create a `MenuBuilder.php` file in `src/Menu`:
+
+``` php hl_lines="20 21 22 26"
+[[= include_file('code_samples/front/layouts/menu/src/Menu/MenuBuilder.php') =]]
+```
+
+In the builder you can define items you want in the menu.
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:51]:
```suggestion
In the builder, you can define items that you want in the menu.
```

File: docs/guide/content_rendering/layout/add_menu.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+For example, lines 20-22 add a specific Location by using the [`ez_urlalias`](../twig_function_reference/url_twig_functions.md#ez_urlalias) route.
+Line 26 adds a defined system route leading to the search form.
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:53]:
```suggestion
Line 26 adds a defined system route that leads to the search form.
```

File: docs/guide/content_rendering/layout/add_menu.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+Next, register the menu builder as a service:
+
+``` yaml
+App\Menu\MenuBuilder:
+    tags:
+        - {name: knp_menu.menu_builder, method: buildMenu, alias: root}
+```
+
+Finally, you can render the menu in `pagelayout.html.twig`.
+Identify it by the name you provided in the Menu Builder's `buildMenu()` method:
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_menu.md:64]:
```suggestion
Identify it by the name that you provided in the Menu Builder's `buildMenu()` method:
```

File: docs/guide/content_rendering/layout/add_search_form.md [hunk 8]
```diff
@@ -8,9 +8,10 @@ and to configure search form and result rendering.
 
 ## Add a search bar
 
-First, include a search bar template in your general layout,
-for example, in the `pagelayout.html.twig` for your theme.
-This example uses the default `standard` theme.
+
+
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_search_form.md:12]:
```suggestion
```

File: docs/guide/content_rendering/layout/add_search_form.md [hunk 9]
```diff
@@ -8,9 +8,10 @@ and to configure search form and result rendering.
...
+First, prepare a general layout template in a `templates/themes/<theme_name>/pagelayout.html.twig` file.
+Include a search bar in this template:
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_search_form.md:14]:
```suggestion
First, prepare a general layout template in a `templates/themes/<theme_name>/pagelayout.html.twig` file, 
and include a search bar in this template:
```

File: docs/guide/content_rendering/layout/add_search_form.md [hunk 10]
```diff
@@ -31,7 +32,7 @@ The `parts/search_bar.html.twig` template uses the built-in `SearchController` t
 You can now go to the front page of your installation.
 An unstyled search bar appears at the top of the page.
 
-## Search result page
+## Customize search result page
 
 Search results are shown using the `/search` route.
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/add_search_form.md:37]: I'd replace "using" with a more specific term here.

File: docs/guide/content_rendering/layout/customize_basket.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
+# Customize basket
+
+To customize the look of the basket, you need to overwrite the default template.
```
Comment (dabrt) [inline docs/guide/content_rendering/layout/customize_basket.md:3]:
```suggestion
To customize the look of the basket, you overwrite the default template.
```
Is it overwrite or override? The next article "overrides" shop templates

**Review by dabrt — APPROVED** [review]

File: docs/guide/content_rendering/layout/add_search_form.md (hunk 10 above)
Comment (DominikaK) [inline docs/guide/content_rendering/layout/add_search_form.md:37]:
```suggestion
Search results are shown in the `/search` route.
```

File: docs/guide/content_rendering/layout/customize_basket.md (hunk 11 above)
Comment (DominikaK) [inline docs/guide/content_rendering/layout/customize_basket.md:3]:
```suggestion
To customize the look of the basket, you override the default template.
```

**Pull request merged**